Add CONTRIBUTING.md (#631)
topolvm/topolvm: README.md

TopoLVM is a [CSI][] plugin using LVM for Kubernetes.
It can be considered as a specific implementation of [local persistent volumes](https://kubernetes.io/docs/concepts/storage/volumes/#local) using CSI and LVM.

- **Project Status**: Testing for production
- **Conformed CSI version**: [1.5.0](https://github.com/container-storage-interface/spec/blob/v1.5.0/spec.md)

> Our supported platform is:

Author: done

*1 The host's Linux Kernel must be v4.9 or later which supports `rmapbt` and `reflink`, if you use xfs filesystem with an official docker image.

> Docker images are available on [ghcr.io](https://github.com/orgs/topolvm/packages)  

Reviewer: ```suggestion
Docker images are available on [ghcr.io](https://github.com/orgs/topolvm/packages).  
```

> **Please note that [the images on quay.io](https://quay.io/organization/topolvm) are currently deprecated and will no longer be available in the future.**

Author: done

Getting started
---------------

A demonstration of TopoLVM running on [kind (Kubernetes IN Docker)][kind] is available at [example](example/) directory.

For production deployments, see [deploy/README.md](./deploy/README.md).

User manual is at [docs/user-manual.md](docs/user-manual.md).

_Deprecated: If you want to use TopoLVM on [Rancher/RKE](https://rancher.com/docs/rke/latest/en/), see [docs/deprecated/rancher.md](docs/deprecated/rancher.md)._

Contributing
------------

> Our project welcomes contributions from any member of our community. To get

Author: done

> started contributing, please see our [Contributor Guide](CONTRIBUTING.md).

Reviewer 2: ```suggestion
started contributing, please see our [Contributing Guide](CONTRIBUTING.md).
```

Author: fixed. (it is from the template though :)

- [Snapshot](https://kubernetes-csi.github.io/docs/snapshot-restore-feature.html): When we want it.

### Deprecated features

> - TopoLVM currently uses [Pod security policy](https://kubernetes.io/docs/concepts/policy/pod-security-policy/), but it has been deprecated since Kubernetes 1.21, and will be deleted in Kubernetes 1.25. So it is also deprecated in TopoLVM.

Reviewer 3: ```suggestion
- [Pod security policy](https://kubernetes.io/docs/concepts/policy/pod-security-policy/) support is deprecated and will be removed when TopoLVM removes the support of Kubernetes < v1.25. This policy is same as Kubernetes.
```

Author: done

Communications
--------------

> If you have any questions or idea to contribute, please use [discussions](https://github.com/topolvm/topolvm/discussions).

Reviewer: ```suggestion
If you have any questions or ideas to contribute, please use [discussions](https://github.com/topolvm/topolvm/discussions).
```

> If you have any questions or ideas to contribute, please use [discussions](https://github.com/topolvm/topolvm/discussions).

Author: done

Resources
---------

> [docs](docs/) directory contains documents about designs and specifications.

Reviewer 3: ```suggestion
[docs](docs/) directory contains the user manual, designs and specifications, and so on.
```

Author: done

A diagram of components is available in [docs/design.md](docs/design.md#diagram).

> The team presented the motivation and implementation of TopoLVM at KubeCon Europe 2020: https://kccnceu20.sched.com/event/ZerD

Reviewer 3: ```suggestion
TopoLVM maintainers presented the motivation and implementation of TopoLVM at KubeCon Europe 2020: https://kccnceu20.sched.com/event/ZerD
```

Author: done
